'no-undef' error when using type annotation of 'HTMLCollectionOf'

## My repogitory
https://github.com/tobigumo/nuxt-eslint-error

This repository is the smallest sample created from create-nuxt-app

## Error details
I've added a type annotation of 'HTMLCollectionOf'.
```
let elements: HTMLCollectionOf<Element>
```
https://github.com/tobigumo/nuxt-eslint-error/blob/361e0542210bc9d0da033bbf6005359b8f94bc2c/pages/index.vue#L33

and I got a 'no-undef' error
```
  # npm run lint
  33:15  error  'HTMLCollectionOf' is not defined                    no-undef
```

## Package versions
```
{
  "dependencies": {
    "@nuxt/typescript-runtime": "^2.0.1",
    "core-js": "^3.8.3",
    "nuxt": "^2.14.12"
  },
  "devDependencies": {
    "@nuxt/types": "^2.14.12",
    "@nuxt/typescript-build": "^2.0.4",
    "@nuxtjs/eslint-config-typescript": "^5.0.0",
    "@nuxtjs/eslint-module": "^3.0.2",
    "babel-eslint": "^10.1.0",
    "eslint": "^7.18.0",
    "eslint-plugin-nuxt": "^2.0.0",
    "eslint-plugin-vue": "^7.5.0"
  }
}
```

--- a/packages/eslint-config-typescript/index.js
+++ b/packages/eslint-config-typescript/index.js
@@ -11,7 +11,7 @@ module.exports = {
     // Per the docs, the root no-unused-vars should be disabled:
     // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unused-vars.md
     'no-unused-vars': 'off',
-    
+
     // https://github.com/typescript-eslint/typescript-eslint/blob/1cf9243/docs/getting-started/linting/FAQ.md#i-get-errors-from-the-no-undef-rule-about-global-variables-not-being-defined-even-though-there-are-no-typescript-errors
     'no-undef': 'off'
   }

--- a/packages/eslint-config-typescript/index.js
+++ b/packages/eslint-config-typescript/index.js
@@ -10,6 +10,9 @@ module.exports = {
     '@typescript-eslint/no-unused-vars': ['error', { args: 'all', argsIgnorePattern: '^_' }],
     // Per the docs, the root no-unused-vars should be disabled:
     // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unused-vars.md
-    'no-unused-vars': 'off'
+    'no-unused-vars': 'off',
+    
+    // https://github.com/typescript-eslint/typescript-eslint/blob/1cf9243/docs/getting-started/linting/FAQ.md#i-get-errors-from-the-no-undef-rule-about-global-variables-not-being-defined-even-though-there-are-no-typescript-errors
+    'no-undef': 'off'
   }
 }

